src/app/models/operation.model.ts

```diff
@@ -1,43 +1,106 @@
+import {Node} from "./node.model";
 import {Trainrun} from "./trainrun.model";
 import {TrainrunSection} from "./trainrunsection.model";
 
 enum OperationType {
   create = "create",
   update = "update",
-  delete = "delete"
+  delete = "delete",
 }
 
-export abstract class Operation {
+enum OperationObjectType {
+  trainrun = "trainrun",
+  trainrunSection = "trainrunSection",
+  trainrunSections = "trainrunSections",
+  node = "node",
+  nodes = "nodes",
```

Comment:
If we have the plural versions, do we need the singular versions? Couldn't we generate a `trainrunSections` operation with an array with a single element instead of `trainrunSection`?

Reply:
agreed, but not used anymore since the design has been simplified